src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -425,6 +439,15 @@ serialize(Archive& ar, const unsigned int /* version */)
   ar & BOOST_SERIALIZATION_NVP(generator);
   ar & BOOST_SERIALIZATION_NVP(discriminator);
   ar & BOOST_SERIALIZATION_NVP(noiseFunction);
+  ar & BOOST_SERIALIZATION_NVP(noiseDim);
+  ar & BOOST_SERIALIZATION_NVP(numFunctions);
+  ar & BOOST_SERIALIZATION_NVP(numDimensions);
+  ar & BOOST_SERIALIZATION_NVP(batchSize);
+  ar & BOOST_SERIALIZATION_NVP(generatorUpdateStep);
+  ar & BOOST_SERIALIZATION_NVP(preTrainSize);
+  ar & BOOST_SERIALIZATION_NVP(multiplier);
+  ar & BOOST_SERIALIZATION_NVP(clippingParameter);
+  ar & BOOST_SERIALIZATION_NVP(lambda);
```

Comment:
Should we call `Reset()` when we are loading?  Otherwise I have some concerns that `predictors` and `responses` could be out of date.

If you want to check whether or not the archive is loading or saving you can use the constexpr boolean values `Archive::is_loading::value` and `Archive::is_saving::value`.

Reply:
@rcurtin I have changed the code in such a way that whenever the model gets loaded, and `Train()` is executed, internally it gives a call to `ResetData()` which initializes the matrix sizes for the predictors and responses, and later calls `Reset()` which initializes the weight of the parameters. So I think, if need be, we can add a call to `ResetData()` here in place of `Reset()`?

Reply:
That seems reasonable, but a user may also call `Predict()` or any other methods right after loading, so we should be sure that the GAN is in a ready-to-use state after serialization.  If all those methods do work, then no problem. :)

You can add a pretty easy serialization test where you train a GAN, then serialize/deserialize and do predictions with it, and make sure the predictions are the same as the original.  Also you could do the same except then also do a little training after deserializing.  Let me know if you want me to clarify, I know I am not being totally precise with what I am writing here :)